cadcdata/cadcdata/core.py

```diff
@@ -0,0 +1,418 @@
+                    self._save_bytes(response, f, decompress, process_bytes)
+            else:
+                self._save_bytes(response, file, decompress, process_bytes)
+        else:
+            # get the file name from the content disposition
+            content_disp = response.headers.get('content-disposition', '')
+            file = file_id
+            for content in content_disp.split():
+                if 'filename=' in content:
+                    file = content[9:]
+            if file.endswith('gz') and decompress:
+                file = os.path.splitext(file)[0]
+            self.logger.debug('Using content disposition file name: {}'.format(file))
+            with open(file, 'w') as f:
+                self._save_bytes(response, f, decompress, process_bytes)
+        self.logger.debug('Successfully saved file\n')
+
+
+    def _save_bytes(self, response, file, decompress=False, process_bytes=None):
+        # requests automatically decompresses the data. Tell it to do it only if it had to
+        if not decompress:
+            chunk = response.raw.read(READ_BLOCK_SIZE)
+            while len(chunk) > 0:
+                if process_bytes is not None:
+                    process_bytes(chunk)
+                file.write(chunk)
+                chunk = response.raw.read(READ_BLOCK_SIZE)
+        else:
+            chunk = response.iter_content(READ_BLOCK_SIZE)
+            while len(chunk) > 0:
+                if process_bytes is not None:
+                    process_bytes(chunk)
+                file.write(chunk)
+                chunk = response.iter_content(READ_BLOCK_SIZE)
+
+
+    def put_file(self, archive, file_id, file, archive_stream=None):
+        """
+        Puts a file into the archive storage
+        :param archive: name of the archive
+        :param file_id: ID of file in the archive storage
+        :param file: location of the source file
+        :param archive_stream: specific archive stream
+        """
+        assert archive is not None
+        assert file_id is not None
+        resource = '/{}/{}'.format(archive, file_id)
+        self.logger.debug('PUT {}'.format(resource))
+        headers = {}
+        if archive_stream is not None:
+            headers[ARCHIVE_STREAM_HTTP_HEADER] = str(archive_stream)
+        with open(file, 'rb') as f:
+            self._repo_client.put(resource, headers=headers, data=f)
+        self.logger.debug('Successfully updated file\n')
+
+
+    def get_file_info(self, archive, file_id):
+        """
+        Get information regarding a file in the archive
+        :param archive: Name of the archive
+        :param file_id: ID of the file
+        :returns dictionary of attributes/values
+        """
+        assert archive is not None
+        assert file_id is not None
+        resource = '/{}/{}'.format(archive, file_id)
+        self.logger.debug('HEAD {}'.format(resource))
+        response = self._repo_client.head(resource)
+        h = response.headers
```

Comment:
This could be clarified significantly if you used a dictionary to express the mapping and then looped over that dictionationary:

file_info_header_map = {'name': 'Content-Disposition', 'size': 'Content-Length' .... }
for key in file_info_header_map:
    file_info[key] = h.get(file_info_header_map[key], None)